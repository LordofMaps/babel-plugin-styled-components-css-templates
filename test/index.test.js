import path from 'path'
import fs from 'fs'
import { transformFileSync } from 'babel-core'
import plugin from '../src'

describe('fixtures', () => {
  const fixturesDir = path.join(__dirname, 'fixtures')
  fs.readdirSync(fixturesDir).sort().map((caseName) => {
    if (caseName === '.DS_Store') return
    it(`should ${caseName.replace(/^\d*-/, '').split('-').join(' ')}`, () => {
      const fixtureDir = path.join(fixturesDir, caseName)
      const fixturePath = path.join(fixtureDir, 'before.js');
      const fixture = transformFileSync(fixturePath).code

      expect(fixture).toMatchSnapshot()
    })
  })
})
