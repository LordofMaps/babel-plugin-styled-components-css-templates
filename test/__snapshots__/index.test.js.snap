// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fixtures should template css helper 1`] = `
"const Simple = css\`#id {width: 100%; } \`;

const Nested = css\`#id {
  width: 100%;

  &:hover {
    color: papayawhip;
  }

  > div {
    background: white;
  }
 } \`;

const Interpolations = css\`#id {
  width: \${props => props.width};
 } \`;

const NestedAndInterpolations = css\`#id {
  width: \${props => props.width};

  &:hover {
    color: \${props => props.color};
  }
 } \`;

const SelectorInterpolation = css\`#id {
  width: \${props => props.width};

  \${props => props.selector} {
    color: papayawhip;
  }
 } \`;

const Styled = styled.div\`&& {width: 100%}\`;"
`;

exports[`fixtures should template inject global 1`] = `
"injectGlobal\`#id > {
  \${glob}

  // comment
  /* comment */
  /*! preserve comment */

  html, body {
    margin: 100000px;
    padding: \${test};

    @media (max-width: 999px) {
      margin: 0;
    }
  }

  .root {
    width: 100%;
  }
}\`;

const Simple = css\`width: 100%;\`;

const Styled = styled.div\`&& {width: 100%}\`;"
`;

exports[`fixtures should template styled 1`] = `
"const Simple = styled.div\`#id && {width: 100%;}\`;

const Nested = styled.div\`#id && {
  width: 100%;

  &:hover {
    color: papayawhip;
  }

  > div {
    background: white;
  }
}\`;

const Interpolations = styled.div\`#id && {
  width: \${props => props.width};
}\`;

const NestedAndInterpolations = styled.div\`#id && {
  width: \${props => props.width};

  &:hover {
    color: \${props => props.color};
  }
}\`;

const SelectorInterpolation = styled.div\`#id && {
  width: \${props => props.width};

  \${props => props.selector} {
    color: papayawhip;
  }
}\`;

const CssHelper = css\`width: 100%\`;"
`;

exports[`fixtures should template styled css inject global 1`] = `
"injectGlobal\`* {
  \${glob}

  // comment
  /* comment */
  /*! preserve comment */

  html, body {
    margin: 100000px;
    padding: \${test};

    @media (max-width: 999px) {
      margin: 0;
    }
  }

  .root {
    width: 100%;
  }
}\`;

const Simple = css\`#id && {width: 100%;}\`;

const Styled = styled.div\`&& {width: 100%}\`;"
`;
